pkg/controller/controller.go

```diff
@@ -689,6 +700,20 @@ func (p *csiProvisioner) Provision(ctx context.Context, options controller.Provi
 		}
 	}
 
+	pvReadOnly := false
+	volCaps := req.GetVolumeCapabilities()
+	// if the request only has one accessmode and if its ROX, set readonly to true
+	if len(volCaps) == 1 {
+		if volCaps[0].GetAccessMode().GetMode() == csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY || volCaps[0].GetAccessMode().GetMode() == csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY {
```

Comment:
csi.VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY is not really mapped to ReadOnlyMany by the external-provisioner, right?  So if you are checking it here, I'd suggest that you add a note to clarify that.

Reply:
Do we ever set SINGLE_NODE_READER_ONLY?

Reply:
Does not look like it is set anywhere: https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L407